RELEASE/scripts/autoscend/combat/auto_combat_wereprofessor.ash

```diff
@@ -29,10 +29,12 @@ string auto_combatWereProfessorStage4(int round, monster enemy, string text)
 		return "";
 	}
 
-	string advresearch = get_property("wereProfessorAdvancedResearch");
-	if(contains_text(advresearch, enemy.id))
+	foreach str in split_string(get_property("wereProfessorAdvancedResearch").to_string(),",")
 	{
-		return "";
+		if(str == enemy.id)
```

Comment:
Actually you're comparing a string to an int here, you need to convert them to the same type